src/test/kotlin/uk/gov/justice/digital/hmpps/prisontonhs/integration/IntegrationTest.kt

```diff
@@ -83,4 +93,12 @@ abstract class IntegrationTest {
 
   internal fun Any.asJson() = gson.toJson(this)
 
+  internal fun setAuthorisation(user: String = "prison-to-nhs-api-client", roles: List<String> = listOf()): (HttpHeaders) -> Unit {
+    val token = jwtHelper.createJwt(subject = user,
+        scope = listOf("read"),
+        expiryTime = Duration.ofHours(1L),
+        roles = roles)
+    return { it.set(HttpHeaders.AUTHORIZATION, "Bearer $token") }
```

Comment:
don't really get this line, what's going on?
could the whole line be replaced with `it.set(HttpHeaders.AUTHORIZATION, "Bearer $token")`?

Reply:
oh right, it's returning a function, that's weird syntax